### PR TITLE
Vet Clinic - Project 4

### PR DESCRIPTION
**Pull request by david-lafontant**
### Vet Clinic - Project 4

**Project requirements**

- Create a table named vets
- Create a many-to-many relationship between the tables species and vets and name it specializations
- Create a many-to-many relationship between the tables animals and vets and name it visits
- Add the data described in this [link](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_join_table.md#:~:text=Insert%20the%20following,Jan%2011th%2C%202021.)
- Add the queries described in this [link]( https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_join_table.md#:~:text=Write%20queries%20to,gets%20the%20most.)

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -63,3 +63,42 @@ UPDATE animals SET owner_id = 2 WHERE name IN ('Gabumon', 'Pikachu');
...
+-- Project 4
+
+INSERT INTO vets (name, age, date_of_graduation)
+VALUES
+  ('William Tatcher', 45, '2000-04-23'),
+  ('Maisy Smith', 26, '2019-01-17'),
+  ('Stephanie Mendez', 64, '1981-05-04'),
+  ('Jack Harkness', 38, '2008-06-08');
+
+INSERT INTO specialization (species_id, vets_id)
```
Comment (nicupop729) [inline data.sql:76]: [No need to action]
At the moment, this is a hard-coded insertion of data in the table. Things can get more complicated than this simple insertion and is better if we have a different approach, like this one, for example:
`INSERT INTO specializations (species_id, vet_id)
 VALUES ((SELECT id FROM vets WHERE name = 'William Tatcher'), (SELECT id FROM species WHERE name = 'Pokemon'));
`

File: data.sql [hunk 2]
```diff
@@ -63,3 +63,42 @@ UPDATE animals SET owner_id = 2 WHERE name IN ('Gabumon', 'Pikachu');
...
+VALUES
+  (1, 1),
+  (1, 3),
+  (2, 3),
+  (2, 4);
+
+INSERT INTO visits (animal_id, vets_id, date_of_visit)
```
Comment (nicupop729) [inline data.sql:83]: [No need to action]
Same here, for the moment this is a hardcoded insertion.
A different approach, the recommendable one is like this:
``INSERT INTO visits (animal_id, vet_id, vist_date) VALUES ((SELECT id FROM animal WHERE name = 'Agumon'), (SELECT id FROM vets WHERE name = 'William Tatcher'), '2020-05-24');``

**Review by nicupop729 — APPROVED** [review]
# STATUS: APPROVED :green_circle: :clap: 

Hi @david-lafontant , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

**Please, take a look at the comments in the code**



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: data.sql (hunk 2 above)
Comment (david-lafontant) [reply data.sql:83]: @nicupop729 Thank you for this tip.

File: data.sql (hunk 1 above)
Comment (david-lafontant) [reply data.sql:76]: @nicupop729 Thank you for this tip.

**Pull request merged**